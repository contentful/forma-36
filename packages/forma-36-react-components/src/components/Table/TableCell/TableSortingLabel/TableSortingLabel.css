@import 'settings/colors';
@import 'settings/transitions';

@import '../TableCell/TableCell.css';

.TableSortingLabel__button {
  appearance: none;
  border: none;
  background: none;
  color: inherit;
  cursor: pointer;
  display: flex;
  font-weight: inherit;
  font-size: inherit;
  text-align: inherit;
  width: 100%;

  &:hover {
    background-color: var(--color-element-light);
  }

  &:active {
    outline: 0;
  }
}

.TableSortingLabel__button__text {
  padding: var(--spacing-s) var(--spacing-m);
  align-items: center;
  width: 100%;
}

.TableSortingLabel__icon {
  transition: transform var(--transition-duration-default)
    var(--transition-easing-default);
  margin: -5px 0;
}

.TableSortingLabel__icon--asc {
  transform: rotateX(0deg);
}

.TableSortingLabel__icon--desc {
  transform: rotateX(180deg);
}
