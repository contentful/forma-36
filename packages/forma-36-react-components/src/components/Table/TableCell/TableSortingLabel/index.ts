export * from './TableSortingLabel';
export { default } from './TableSortingLabel';
