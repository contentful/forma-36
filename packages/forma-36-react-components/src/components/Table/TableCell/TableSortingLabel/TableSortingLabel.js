import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import { iconName } from '../../../Icon/constants';
import Icon from '../../../Icon/Icon';
import { sortingDirections } from '../TableCell/TableCell';
import TabFocusTrap from '../../../TabFocusTrap';
import styles from './TableSortingLabel.css';

export default class TableSortingLabel extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node.isRequired,
    direction: PropTypes.oneOf(Object.keys(sortingDirections)).isRequired,
    active: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    extraClassNames: '',
  };

  renderIcon() {
    const { direction } = this.props;
    const classNames = cn(
      styles.TableSortingLabel__icon,
      styles[`TableSortingLabel__icon--${direction}`],
    );

    return (
      <Icon
        extraClassNames={classNames}
        icon={iconName.ArrowUp}
        color="muted"
      />
    );
  }

  render() {
    const { extraClassNames, children, active, ...otherProps } = this.props;

    return (
      <button
        type="button"
        className={cn(styles.TableSortingLabel__button, extraClassNames)}
        {...otherProps}
      >
        <TabFocusTrap extraClassNames={styles.TableSortingLabel__button__text}>
          {children}
          {active && this.renderIcon()}
        </TabFocusTrap>
      </button>
    );
  }
}
