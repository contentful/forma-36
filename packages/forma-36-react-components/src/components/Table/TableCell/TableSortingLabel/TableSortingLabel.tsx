import React, { ButtonHTMLAttributes } from 'react';
import cn from 'classnames';

import Icon from '../../../Icon/Icon';
import { sortingDirections } from '../TableCell/TableCell';
import TabFocusTrap from '../../../TabFocusTrap/TabFocusTrap';
import styles from './TableSortingLabel.css';

export interface TableSortingLabelProps
  extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  direction: keyof typeof sortingDirections;
  active: boolean;
  className?: string;
  testId?: string;
}

const defaultProps: Partial<TableSortingLabelProps> = {
  testId: 'cf-ui-table-sorting-label',
};

const TableSortingLabel = (props: TableSortingLabelProps) => {
  const { className, children, active, testId, ...otherProps } = props;

  const renderIcon = () => {
    const { direction } = props;
    const classNames = cn(
      styles['TableSortingLabel__icon'],
      styles[`TableSortingLabel__icon--${direction as string}`],
    );

    return <Icon className={classNames} icon="ArrowUp" color="muted" />;
  };

  return (
    <button
      type="button"
      className={cn(styles['TableSortingLabel__button'], className)}
      data-test-id={testId}
      {...otherProps}
    >
      <TabFocusTrap className={styles['TableSortingLabel__button__text']}>
        {children}
        {active && renderIcon()}
      </TabFocusTrap>
    </button>
  );
};

TableSortingLabel.defaultProps = defaultProps;

export default TableSortingLabel;
