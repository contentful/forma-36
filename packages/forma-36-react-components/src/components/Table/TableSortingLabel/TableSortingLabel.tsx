import React from 'react';
import cn from 'classnames';
import type { ButtonHTMLAttributes } from 'react';
import { ArrowUp } from '@contentful/f36-icons';

import { sortingDirections } from '../TableCell/TableCell';
import { TabFocusTrap } from '@contentful/f36-utils';
import styles from './TableSortingLabel.css';

export interface TableSortingLabelProps
  extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  direction: keyof typeof sortingDirections;
  active: boolean;
  className?: string;
  testId?: string;
}

export const TableSortingLabel = ({
  active,
  children,
  className,
  direction,
  testId = 'cf-ui-table-sorting-label',
  ...otherProps
}: TableSortingLabelProps) => {
  const renderIcon = () => {
    const classNames = cn(
      styles['TableSortingLabel__icon'],
      styles[`TableSortingLabel__icon--${direction as string}`],
    );

    return <ArrowUp className={classNames} variant="muted" />;
  };

  return (
    <button
      type="button"
      className={cn(styles['TableSortingLabel__button'], className)}
      data-test-id={testId}
      {...otherProps}
    >
      <TabFocusTrap className={styles['TableSortingLabel__button__text']}>
        {children}
        {active && renderIcon()}
      </TabFocusTrap>
    </button>
  );
};
