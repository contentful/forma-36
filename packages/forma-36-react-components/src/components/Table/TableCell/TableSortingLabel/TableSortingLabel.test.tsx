import React from 'react';
import { shallow } from 'enzyme';
import TableSortingLabel from './TableSortingLabel';

it('renders the component', () => {
  const output = shallow(
    <TableSortingLabel active direction="asc">
      Foo
    </TableSortingLabel>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component inactive', () => {
  const output = shallow(
    <TableSortingLabel active={false} direction="desc">
      Foo
    </TableSortingLabel>,
  );

  expect(output).toMatchSnapshot();
});
