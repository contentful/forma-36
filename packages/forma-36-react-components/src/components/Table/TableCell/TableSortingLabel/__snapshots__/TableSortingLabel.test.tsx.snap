// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<button
  class="TableSortingLabel__button"
  data-test-id="cf-ui-table-sorting-label"
  direction="asc"
  type="button"
>
  <span
    class="TabFocusTrap TableSortingLabel__button__text"
    tabindex="-1"
  >
    Foo
    <svg
      class="Icon Icon--small Icon--muted TableSortingLabel__icon TableSortingLabel__icon--asc"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M7 14l5-5 5 5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</button>
`;

exports[`renders the component inactive 1`] = `
<button
  class="TableSortingLabel__button"
  data-test-id="cf-ui-table-sorting-label"
  direction="desc"
  type="button"
>
  <span
    class="TabFocusTrap TableSortingLabel__button__text"
    tabindex="-1"
  >
    Foo
  </span>
</button>
`;
